pkg/controller/association/reconciler_test.go

```diff
@@ -456,46 +455,197 @@ func TestReconciler_Reconcile_NewAssociation(t *testing.T) {
 	// association status should be established
 	require.NoError(t, err)
 	require.Equal(t, commonv1.AssociationEstablished, updatedKibana.Status.AssociationStatus)
 }
 
+func TestReconciler_Reconcile_noESAuth(t *testing.T) {
+	// Kibana references Enterprise Search, the association controller is configured to not
+	// create an Elasticsearch user
+	ent := entv1.EnterpriseSearch{ObjectMeta: metav1.ObjectMeta{Namespace: "entns", Name: "entname"}}
+	entHTTPPublicCertsSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "entns",
+			Name:      "entname-ent-http-certs-public",
+		},
+		Data: map[string][]byte{
+			"ca.crt":  []byte("ca cert content"),
+			"tls.crt": []byte("tls cert content"),
+		},
+	}
+	entHTTPService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "entns",
+			Name:      "entname-ent-http",
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name: "https",
+					Port: 3002,
+				},
+			},
+		},
+	}
+	kb := sampleKibanaNoEsRef()
+	kb.Spec.EnterpriseSearchRef = commonv1.ObjectSelector{Name: "entname", Namespace: "entns"}
+
+	// ent public certs we expect to be copied over into the Kibana namespace
+	entCertsInKibanaNamespace := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: kibanaNamespace,
+			Name:      "kbname-kb-ent-ca",
+			Labels: map[string]string{
+				"enterprisesearch.k8s.elastic.co/name":       "entname",
+				"enterprisesearch.k8s.elastic.co/namespace":  "entns",
+				"kibanaassociation.k8s.elastic.co/name":      "kbname",
+				"kibanaassociation.k8s.elastic.co/namespace": "kbns",
+				"kibanaassociation.k8s.elastic.co/type":      "ent",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "kibana.k8s.elastic.co/v1",
+					Kind:               "Kibana",
+					Name:               "kbname",
+					Controller:         &varTrue,
+					BlockOwnerDeletion: &varTrue,
+				},
+			},
+		},
+		Data: map[string][]byte{
+			"ca.crt":  []byte("ca cert content"),
+			"tls.crt": []byte("tls cert content"),
+		},
+	}
+
+	kbEntAssocInfo := AssociationInfo{
+		AssociatedObjTemplate: func() commonv1.Associated { return &kbv1.Kibana{} },
+		ReferencedObjTemplate: func() client.Object { return &entv1.EnterpriseSearch{} },
+		ExternalServiceURL: func(c k8s.Client, assoc commonv1.Association) (string, error) {
+			entRef := assoc.AssociationRef()
+			if !entRef.IsDefined() {
+				return "", nil
+			}
+			ent := entv1.EnterpriseSearch{}
+			if err := c.Get(context.Background(), entRef.NamespacedName(), &ent); err != nil {
+				return "", err
+			}
+			serviceName := entRef.ServiceName
+			if serviceName == "" {
+				serviceName = "entname-ent-http"
+			}
+			nsn := types.NamespacedName{Namespace: ent.Namespace, Name: serviceName}
+			return ServiceURL(c, nsn, ent.Spec.HTTP.Protocol())
```

Comment:
What's the reason for not just using `getEntExternalServiceURL` here?

Reply:
Circular dependencies :(
the association package is supposed to be generic to any association controller defined in `association/controllers` (which itself imports `association/`).
Here we test the "generic" reconciler logic, not the kibana<->ent association controller specifically.
Since it's for unit tests only I was thinking a bit of code duplication is not _too_ bad (we could also make this function much simpler in the unit test).

Reply:
Makes sense!